### ISSUE TITLE
minor mistakes in 1st edition 

### ISSUE THREAD

**Issue by kuba-04** [braiins/build-bitcoin-in-rust#6]
not a big deal, I believe this is just a fresh 1st edition so some mistakes are understandable. 

A "bug" in the book:
page 213 we are adding a block of code:
`block.verify_transactions(
  self.block_height(),
  &self.utxos,
)?;`

then on the page 217 the book says:
"you should have not just no errors, but also no warnings"

but it can't compile because the above method will be added just when we are reading further! 

I think I have spotted similar but smaller things before but didn't note them down. Do  you want me to add such notes here under this thread, so it will be easier to correct things for the second edition? 

Great book btw! I really enjoy following it. Thank you @luciusmagn for this effort

**Comment by kuba-04** [comment]
again: page 226
`self.mempool`
we never added mempool, the previous page shows clearly Blockchain fields as of now:
`utxos, target, blocks`
so mempool wasn't added and we have to comment out to compile.

**Comment by killmasud** [comment]
Hey, i bumped into the same thing, but i thought the author was making a cheeky joke because of the ":)" smiley face. Otherwise I can confirm the same.

**Comment by arejula27** [comment]
Yep, I spent time looking for the function implementation in the previous pages, definitely not convenient, haha
